internal/tasks/schedulertask.go

```diff
@@ -202,132 +223,172 @@ func (sm *SchedulerTask) Run(ctx context.Context) error {
 	return err
 }
 
+func (sm *SchedulerTask) refreshCache(ctx context.Context) error {
+	topics, terr := sm.topicCore.List(ctx, topic.Prefix)
+	if terr != nil {
+		logger.Ctx(ctx).Errorw("error fetching topic list", "error", terr)
+		return terr
+	}
+
+	topicData := make(map[string]*topic.Model)
+	for _, topic := range topics {
+		topicData[topic.Name] = topic
+	}
+	sm.topicCache = topicData
+
+	nodes, nerr := sm.nodeCore.List(ctx, node.Prefix)
+	if nerr != nil {
+		logger.Ctx(ctx).Errorw("error fetching new node list", "error", nerr)
+		return nerr
+	}
+
+	nodeData := make(map[string]*node.Model)
+	for _, node := range nodes {
+		nodeData[node.ID] = node
+	}
+	sm.nodeCache = nodeData
+
+	allSubs, serr := sm.subscriptionCore.List(ctx, subscription.Prefix)
+	if serr != nil {
+		logger.Ctx(ctx).Errorw("error fetching new subscription list", "error", serr)
+		return serr
+	}
+	// Filter Push Subscriptions
+	newSubs := make(map[string]*subscription.Model)
+	for _, sub := range allSubs {
+		if sub.IsPush() {
+			newSubs[sub.Name] = sub
+		}
+	}
+
+	sm.subCache = newSubs
+	return nil
+}
+
+// refreshNodeBindings achieves the following in the order outline:
+// 1. Go through node bindings and remove invalid bindings
+// 	 (Invalid due to changes in the subscription, node failures, topic updates, etc)
+//	  a. Remove bindings that do not conform to the new partition based approach.
+//    b. Remove nodebindings for deleted/invalid subscriptions.
+//    c. Remove nodebindings impacted by node failures
+// 2. Evaluate subscriptions and schedule any missing subscription/partition combos to nodes available.
+// 3. Topic changes are inherently covered since subscription validates against topic.
 func (sm *SchedulerTask) refreshNodeBindings(ctx context.Context) error {
+	err := sm.refreshCache(ctx)
```

Comment:
why are we refreshing all three caches, if one is changed..?

Reply:
The existing local cache might still be the one that holds non-partition-specific bindings due to the old code flow. We want to fetch the cache for all three entities and verify that the old bindings are removed and new bindings are added accordingly.